src/qutip_qip/circuit/mat_renderer.py

```diff
@@ -0,0 +1,863 @@
+                        gate.color,
+                    )
+
+            self.ax.add_artist(gate_text)
+            self.ax.add_patch(gate_patch)
+            self._manage_layers(gate_width, wire_list, layer, xskip)
+
+            return None
+
+    def _draw_measure(self, c_pos, q_pos, layer):
+        """
+        Draw the measurement gate.
+
+        Parameters
+        ----------
+        c_pos : int
+            The position of the classical wire.
+
+        q_pos : int
+            The position of the quantum wire.
+
+        layer : int
+            The layer the gate is acting on.
+        """
+
+        xskip = self._get_xskip(
+            list(range(0, self.merged_qubits[-1] + 1)), layer
+        )
+        measure_box = FancyBboxPatch(
+            (
+                xskip + self.gate_margin,
+                (q_pos + self.cwires) * self.wire_sep - self.gate_height / 2,
+            ),
+            self.gate_width,
+            self.gate_height,
+            boxstyle="round4",
+            mutation_scale=0.3,
+            facecolor="white",
+            edgecolor="k",
+            zorder=2,
+        )
+        arc = Arc(
+            (
+                xskip + self.gate_margin + self.gate_width / 2,
+                (q_pos + self.cwires) * self.wire_sep - self.gate_height / 2,
+            ),
+            self.gate_width * 1.5,
+            self.gate_height * 1,
+            angle=0,
+            theta1=0,
+            theta2=180,
+            color="k",
+            zorder=2,
+        )
+        arrow = FancyArrow(
+            xskip + self.gate_margin + self.gate_width / 2,
+            (q_pos + self.cwires) * self.wire_sep - self.gate_height / 2,
+            self.gate_width * 0.7,
+            self.gate_height * 0.7,
+            length_includes_head=True,
+            head_width=0,
+            color="k",
+            zorder=2,
+        )
+
+        self._draw_cbridge(c_pos, q_pos, xskip, "k")
+        self._manage_layers(
+            self.gate_width,
+            list(range(0, self.merged_qubits[-1] + 1)),
+            layer,
+            xskip,
+        )
+        self.ax.add_patch(measure_box)
+        self.ax.add_artist(arc)
+        self.ax.add_artist(arrow)
+
+    def canvas_plot(self):
+        """
+        Plot the quantum circuit.
+        """
+
+        self._extend_wires(self.default_layers)
+        self._add_wire_labels()
+
+        for gate in self.qc.gates:
+
+            # multi-qubit gate
+            if (
+                len(gate.targets) > 1
+                or getattr(gate, "controls", False) is not None
+            ):
+
+                self.merged_qubits = gate.targets.copy()
+                if isinstance(gate, Gate) and gate.controls is not None:
+                    self.merged_qubits += gate.controls.copy()
+                self.merged_qubits.sort()
+
+                # check if the gate is a measurement gate
+                if isinstance(gate, Measurement):
+
+                    find_layer = [
```

Comment:
Why is this list named `find_layer`? Is this a list of the wires that this gate includes?

Reply:
I should rename that. It is a list of layers up to which the plotting is done in the wire; for all the wires that the new gate is acting on.